### PR TITLE
Added some unit tests to improve test coverage.

### LINKED ISSUE #20: Problem connecting to the KBase auth server
Currently, [KBase's Auth server](https://github.com/kbase/auth2) reads a non-standard HTTP Authorization header when it decides whether to allow access to a resource. This hasn't mattered much to date, but Go's HTTP libraries perform client-side HTTP header validation, preventing non-standard headers from being used in requests. We can work around this issue for now in local (development) DTS instances, but it currently prevents us from deploying DTS into any environment where researchers can request file transfers.

### PR DESCRIPTION
**Pull request by jeff-cohere**
Also, I rearranged the structure of the endpoint configuration data to make it more similar to databases.

In testing the JDP Database implementation, I've figured out that the JDP cannot support one of our requirements, which is the ability to return metadata for files with specific identifiers. We need this in order to look up the locations of files for staging.

UPDATE: This branch contains a fix for the KBase Auth server header validation problem.

Closes #20

### CODE REVIEW

File: databases/databases_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"testing"
+)
+
+func TestNewJDPDatabase(t *testing.T) {
+	assert := assert.New(t) // binds assert to t
+	orcid := os.Getenv("DTS_KBASE_TEST_ORCID")
+	jdpDb, err := NewDatabase(orcid, "jdp")
+	assert.NotNil(jdpDb, "JDP database not created")
+	assert.Nil(err, "JDP database creation encountered an error")
+}
```
Comment (ialarmedalien) [inline databases/databases_test.go:15]: Can you add in a quick test for failure? e.g. `NewDatabase(orcid, "cgi")`

File: databases/jdp/database.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ import (
 	"net/url"
 	"os"
 	"path/filepath"
+	//	"slices"
```
Comment (ialarmedalien) [inline databases/jdp/database.go:34]: rm line?

File: endpoints/globus/endpoint_test.go [hunk 3]
```diff
@@ -100,7 +100,7 @@ func TestGlobusStatus(t *testing.T) {
 	assert.NotNil(err)
 }
 
-// This runs setup, runs all tests, and does breakdown.
+// this runs setup, runs all tests, and does breakdown
```
Comment (ialarmedalien) [inline endpoints/globus/endpoint_test.go:103]: Vital capitalisation change!

File: services/prototype.go [hunk 4]
```diff
@@ -9,7 +9,7 @@ import (
 	"log"
 	"net"
 	"net/http"
-	"os"
+	//	"os"
```
Comment (ialarmedalien) [inline services/prototype.go:12]: rm line?

File: endpoints/globus/endpoint_test.go (hunk 3 above)
Comment (jeff-cohere) [reply endpoints/globus/endpoint_test.go:103]: This is mainly to remind me and my OCD that comments are not formal prose, but it must look bizarre!

File: databases/jdp/database_test.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	assert := assert.New(t) // binds assert to t
+	orcid := os.Getenv("DTS_KBASE_TEST_ORCID")
+	jdpDb, err := NewDatabase(orcid)
+	assert.NotNil(jdpDb, "JDP database not created")
+	assert.Nil(err, "JDP database creation encountered an error")
+}
+
+func TestNewDatabaseWithoutOrcid(t *testing.T) {
+	assert := assert.New(t) // binds assert to t
+	jdpDb, err := NewDatabase("")
```
Comment (ialarmedalien) [inline databases/jdp/database_test.go:51]: Does go have the concept of null / undefined, and if so, is it worth adding tests where `orcid` is null/undef?

Comment (jeff-cohere) [reply databases/jdp/database_test.go:51]: Go does have `nil`, but it doesn't have `nil` strings (only blank ones).

Comment (jeff-cohere) [reply databases/jdp/database_test.go:51]: Btw, strings and "byte slices" (`[]byte`) are cousins in GoLand, and all slices do a have a `nil`able representation.

File: databases/jdp/database_test.go [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	assert.NotNil(err, "JDP database creation without ORCID encountered no error")
+}
+
+func TestNewDatabaseWithoutJDPSharedSecret(t *testing.T) {
+	assert := assert.New(t) // binds assert to t
+	orcid := os.Getenv("DTS_KBASE_TEST_ORCID")
+	jdpSecret := os.Getenv("DTS_JDP_SECRET")
+	os.Unsetenv("DTS_JDP_SECRET")
+	jdpDb, err := NewDatabase(orcid)
+	os.Setenv("DTS_JDP_SECRET", jdpSecret)
```
Comment (ialarmedalien) [inline databases/jdp/database_test.go:62]: will the `DTS_JDP_SECRET` be set again if `NewDatabase` throws an error / fails in some way? i.e. will `DTS_JDP_SECRET` always get re-set?

Comment (jeff-cohere) [reply databases/jdp/database_test.go:62]: Yes, it always gets reset, because we reset it before acting on any error conditions. In departing from exception-based error flows, Go (like C) makes one respond to error codes instead. A lot of folks don't like this, but in my experience, there's no way to do error handling that is simultaneously flexible (handle your own errors the way you think they should be handled, when you think they should be handled!) and seamless (toss an exception into the ether and forget about it, and be prepared to catch these things downstream!).

Comment (ialarmedalien) [reply databases/jdp/database_test.go:62]: Nice. Setting/unsetting env vars in python tests reliably requires a lot more fiddling so it's nice to see how much easier it is here!

File: endpoints/globus/endpoint.go [hunk 7]
```diff
@@ -68,8 +68,11 @@ type Endpoint struct {
 }
 
 func NewEndpoint(endpointName string) (core.Endpoint, error) {
-	epConfig, found := config.Globus.Endpoints[endpointName]
+	epConfig, found := config.Endpoints[endpointName]
 	if !found {
+		return nil, fmt.Errorf("'%s' is not an endpoint", endpointName)
+	}
+	if epConfig.Provider != "globus" {
```
Comment (ialarmedalien) [inline endpoints/globus/endpoint.go:75]: are there tests for these two errors? (I love me some comprehensive error testing!)

Comment (jeff-cohere) [reply endpoints/globus/endpoint.go:75]: There are now! :-)

**Review by ialarmedalien — APPROVED** [review]
LGTM!
